### PR TITLE
DBO 58 - Look into creating component tests for major frontend components

### PR DESCRIPTION
**Pull request by sahilsaini432**
- Added Tests for Post Component
- Added Tests for NavBar Component
- Added Tests for TagButtonGroup Component
- Added Tests for CopyButton Component

### CODE REVIEW

File: application/src/components/post/Post.test.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  date_created: '2021-11-20T17:31:03.914+00:00',
+  // date_created: Date.now(),
+  location: {
+    lat: 45.963589,
+    lng: -66.643112
+  },
+  location_string: 'New Brunswick, Canada',
+}
+
+describe('Post Component', () => {
```
Comment (fredy20501) [inline application/src/components/post/Post.test.js:31]: Another test for this component could be to verify that tags are displayed in sorted order. So you could give the tags array in postData in a non-alphabetical order then check that the values for the tag's `div` in the `tag-container` are in alphabetical order.

This is just a small thing though, so don't worry about it if it is difficult/complex to do.

File: application/src/components/CopyButton.test.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    render(<CopyButton value={value} />);
+
+    const copyButton = screen.getByTestId('copy-button');
+    expect(copyButton).toBeInTheDocument();
+    expect(navigator.clipboard.writeText).toBeCalledTimes(0);
+  });
+
+  test('click the copy button', () => {
+    const useStateSpy = jest.spyOn(React, 'useState');
+    useStateSpy.mockImplementation((init) => [init, jest.fn()]);
```
[the remark this answers is not in the document]
Comment (sahilsaini432) [reply application/src/components/CopyButton.test.js:27]: Ya, the test will pass without it, but sometimes it gives this warning without these lines

Screen Shot 2022-02-07 at 12 06 46 PM

**Review by fredy20501 — APPROVED** [review]
Well done! 🥇

**Review by CarterMacLennan — APPROVED** [review]
Well done, looks great to me :)

**Review by JoaoPontes98 — APPROVED** [review]

**Pull request merged**